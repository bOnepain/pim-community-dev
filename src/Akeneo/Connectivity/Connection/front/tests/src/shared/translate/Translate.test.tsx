import * as React from 'react';
import {create} from 'react-test-renderer';
import {Translate} from '@src/shared/translate/Translate';
import {TranslateContext} from '@src/shared/translate/translate-context';

describe('Translate', () => {
    it('should display the translation', () => {
        const translate = jest.fn().mockReturnValue('No product.');

        const component = create(
            <TranslateContext.Provider value={translate}>
                <Translate id='hello.world' />
            </TranslateContext.Provider>
        );

        expect(translate).toBeCalledWith('hello.world', {}, 1);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should handle pluralization and placeholders', () => {
        const translate = jest.fn().mockReturnValue('9 products');

        const component = create(
            <TranslateContext.Provider value={translate}>
                <Translate id='hello.world' count={9} placeholders={{count: '9'}} />
            </TranslateContext.Provider>
        );

        expect(translate).toBeCalledWith('hello.world', {count: '9'}, 9);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
