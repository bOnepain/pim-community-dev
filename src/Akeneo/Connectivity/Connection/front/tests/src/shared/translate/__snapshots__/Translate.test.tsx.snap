// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate should display the translation 1`] = `"No product."`;

exports[`Translate should handle pluralization and placeholders 1`] = `"9 products"`;
